google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -149,7 +151,7 @@ public CancellableRunnable createDeployRunner(
 
     boolean isFlexCompat = targetEnvironment.isFlexible()
         && AppEngineProjectService.getInstance().isFlexCompat(project, source);
-    if (targetEnvironment.isStandard() || isFlexCompat) {
```

Comment:
Standard deployment runner will be used by modules with the standard facet, idem for Flexible. The facet will be the only source of truth.